src/overview/container.js

```diff
@@ -28,6 +32,40 @@ class OverviewContainer extends Component {
         needsWaypoint: PropTypes.bool.isRequired,
         handleTrashBtnClick: PropTypes.func.isRequired,
         handleToggleBm: PropTypes.func.isRequired,
+        pageIdForTag: PropTypes.string.isRequired,
+        handleTagBtnClick: PropTypes.func.isRequired,
+        newTag: PropTypes.string.isRequired,
+        onTagSearchChange: PropTypes.func.isRequired,
+        resultTags: PropTypes.arrayOf(PropTypes.object).isRequired,
+        addTags: PropTypes.func.isRequired,
+        delTags: PropTypes.func.isRequired,
+        onTagSearchEnter: PropTypes.func.isRequired,
+        suggestedTags: PropTypes.arrayOf(PropTypes.string).isRequired,
+        emptyTagOptions: PropTypes.bool.isRequired,
+        hoveredTagResult: PropTypes.string.isRequired,
+        changeHoveredTag: PropTypes.func.isRequired,
+        tagSearchValue: PropTypes.string.isRequired,
+        filterTag: PropTypes.func.isRequired,
+        tagExpandedPageId: PropTypes.string.isRequired,
+        tagExpand: PropTypes.func.isRequired,
+    }
+
+    constructor() {
+        super()
+
+        this.handleOutsideClick = this.handleOutsideClick.bind(this)
+        this.handleKeyBoardDown = this.handleKeyBoardDown.bind(this)
+        this.setTagInputFocus = this.setTagInputFocus.bind(this)
+        this.handleTagEnter = this.handleTagEnter.bind(this)
+        this.handleTagBtnClickAndFocus = this.handleTagBtnClickAndFocus.bind(
+            this,
+        )
+    }
+
+    componentWillMount() {
+        document.addEventListener('click', this.handleOutsideClick, false)
```

Comment:
These three event listeners should be added to the relevant components in the shadow DOM (i.e., the JSX code) rather than the root of the real DOM `document`. If you find out where they should go just use the `on$EVENT_NAME` props on that JSX element to bind the listeners (https://reactjs.org/docs/events.html), same as you already do with things like `onChange`.